v03_pipeline/lib/annotations/reference_dataset_collection.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from typing import Any
+
+import hail as hl
+
+from v03_pipeline.lib.model import (
+    DatasetType,
+    Env,
+    ReferenceDatasetCollection,
+    ReferenceGenome,
+)
+from v03_pipeline.lib.paths import reference_dataset_collection_path
+
+
+def hgmd(
+    ht: hl.Table,
+    env: Env,
+    reference_genome: ReferenceGenome,
+    dataset_type: DatasetType,
+    **_: Any,
+) -> hl.Expression:
+    if (
```

Comment:
This feels like you've copy-pasted the checking in all the functions here. Maybe make `reference_dataset_collection_path` into `valid_reference_dataset_collection_path` and have the helper function handle the checking?

Reply:
Yep!  This is bad copy-paste 🤔 .  

I think the intention here was to make the check very explicit, but it's probably unnecessary and over-engineered.  I believe `reference_dataset_collection_path` already does the check and will return `None` for hgmd in the wrong environment, so I can just use that.  The rename to `valid_reference_dataset_collection_path` is probably explicit enough.